server/controllers/notification.js

```diff
@@ -0,0 +1,90 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+  const userId = req.body.userId;
+  let notifications;
+  if (userId) {
+    notifications = await Notification.find({
+      recipient: userId,
+    })
+      .populate({ path: "owner" })
+      .populate({ path: "recipient" });
+  }
+
+  if (!notifications) {
+    res.status(400);
+    throw new Error("No notification found.");
+  }
+
+  res.status(200).json({ notifications: notifications });
+});
+
+exports.getAllunreadNotifications = asyncHandler(async (req, res, next) => {
+  const userId = req.body.userId;
+  let notifications;
+  if (userId) {
+    notifications = await Notification.find({
+      recipient: userId,
+      read: false,
+    })
+      .populate({ path: "owner" })
+      .populate({ path: "recipient" });
+  }
+
+  if (!notifications) {
+    res.status(400);
+    throw new Error("No notification found.");
+  }
+
+  res.status(200).json({ notifications: notifications });
+});
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const { type, ownerId, recipientId, title, description } = req.body;
+
+  if (!type || !ownerId || !recipientId || !title || !description) {
+    res.status(400);
+    throw new Error("Incomplete required data");
+  }
+
+  let anchor;
+  if (type === "message") {
+    anchor = "message";
+  } else {
+    anchor = "notification center";
+  }
```

Comment:
Is anchor event necessary if we have type already defined?

Reply:
Actually, with the scope of this project, we do not need anchor if we have type.  Thank you for pointing this out.